algorithms/a3c/test.py

```diff
@@ -9,36 +9,49 @@
 import time
 from collections import deque
 
+import numpy as np
 import torch
 import torch.nn.functional as F
 
 from gym_ai2thor.envs.ai2thor_env import AI2ThorEnv
-from algorithms.a3c.envs import create_atari_env
-from algorithms.a3c.model import ActorCritic
+from gym_ai2thor.task_utils import unpack_state
+from algorithms.a3c.env_atari import create_atari_env
+from algorithms.a3c.model import ActorCritic, A3C_LSTM_GA
 
 
 def test(rank, args, shared_model, counter):
     torch.manual_seed(args.seed + rank)
 
     if args.atari:
         env = create_atari_env(args.atari_env_name)
+    elif args.vizdoom:
+        # many more dependencies required for VizDoom
+        from algorithms.a3c.env_vizdoom import GroundingEnv
+
+        env = GroundingEnv(args)
+        env.game_init()
     else:
-        args.config_dict = {'max_episode_length': args.max_episode_length}
-        env = AI2ThorEnv(config_dict=args.config_dict)
+        env = AI2ThorEnv(config_file=args.config_file_path, config_dict=args.config_dict)
     env.seed(args.seed + rank)
 
-    model = ActorCritic(env.observation_space.shape[0], env.action_space.n, args.frame_dim)
+    if env.task.task_has_language_instructions:
```

Comment:
Could we have a convention of tasks with language starting with "Language" or sth like that? so we would check if env.task.__name__.startswith("Language"). That would look better IMO

Reply:
I do have `NaturalLanguage` at the start of all of them and `NL` at the start of the config files. But a boolean is just easier no??

Reply:
and isn't sticking to a standard a good thing?

Reply:
it's added automatically in the NaturalLanguageBaseTask? 
0 work, super clear, very readable, very DRY and doesn't require us to forcibly stick to a standard as we create 100+ tasks